presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -1372,6 +1316,88 @@ public AnalyzePropertyManager getAnalyzePropertyManager()
         return analyzePropertyManager;
     }
 
+    @Override
+    public MetadataResolver getMetadataResolver(Session session)
+    {
+        return new MetadataResolver()
```

Comment:
Since this class is already quite long, it would be preferable to pull this class out to an outer class.

Reply:
I am not sure if it will help with bloat or create more bloat. We will need to pass all used member variables in that case to the Impl class? WDYT

Reply:
Actually thinking about more, it makes sense to move it to outer class. As this class/interface will continue to grow more.  I will make changes.